train.py

```diff
@@ -101,18 +115,39 @@
 logger.info("candidate count: %d", len(candidates))
 logger.info("Running fit() for model %s", model.__class__.__name__)
 
-# Start fitting the model
-fit(
-    epochs=args.epoch_num,
-    model=model,
-    opt=optimizer,
-    train_data=train_data,
-    test_data=test_data,
-    dataset=dataset,
-    candidates=candidates,
-    top_k=args.top_k,
-    batch_size=args.batch_size,
-    save_model_dir=save_model_dir
-)
 
+if args.command == "precompute-embeddings":
```

Comment:
if/else is okay for just two commands, but let's use separate functions if this grows any more